lib/ansible/module_utils/urls.py

```diff
@@ -218,9 +220,109 @@ def __init__(self, message, import_traceback, module=None):
         self.module = module
 
 
+class TLSinTLS:
+    def __init__(self, socket, context, server_hostname):
+        self.incoming = ssl.MemoryBIO()
+        self.outgoing = ssl.MemoryBIO()
+
+        self.socket = socket
+
+        self.sslobj = sslobj = context.wrap_bio(
+            self.incoming, self.outgoing, server_hostname=server_hostname
+        )
+
+        self.read = functools.partial(self._process, True, sslobj.read)
+        self.recv = functools.partial(self._process, True, sslobj.read)
+        self.send = functools.partial(self._process, False, sslobj.write)
+        self.sendall = functools.partial(self._process, False, sslobj.write)
```

Comment:
Do we need to worry about something calling `sendall` for this when the payload exceeds 16KiB? Is that even something we can do, i.e. a POST request with a large file?

Reply:
Ah just realized this is for TLS in TLS so it should never be sending any packets larger than this anyway.